lib/sqlalchemy/cyextension/resultproxy.pyx

```diff
@@ -68,6 +73,13 @@ cdef class BaseRow:
         return self._data[index]
 
     cpdef _get_by_key_impl_mapping(self, key):
+        return self._real_get_by_key_impl_mapping(key)
+
+    cdef _real_get_by_key_impl_mapping(self, key):
```

Comment:
does this make a lot of difference compared just using the cpdef fn?

Reply:
That shouldn't be the bulk of the improvement.  Getting rid of the `try` helps more I think

`cythonize -X language_level=3 -a -i  lib/sqlalchemy/cyextension/resultproxy.pyx`

Before:
[resultproxy.html.zip](https://github.com/sqlalchemy/sqlalchemy/files/11259648/resultproxy.html.zip)

After:
[resultproxy.html.zip](https://github.com/sqlalchemy/sqlalchemy/files/11259642/resultproxy.html.zip)